tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/strategy/ProgrammeBasedFilterStrategy.java

```diff
@@ -3,33 +3,80 @@
 import com.transformuk.hee.tis.security.model.Programme;
 import com.transformuk.hee.tis.security.model.UserProfile;
 import com.transformuk.hee.tis.security.util.TisSecurityHelper;
+import com.transformuk.hee.tis.tcs.api.enumeration.ProgrammeMembershipStatus;
+import com.transformuk.hee.tis.tcs.service.model.ColumnFilter;
+import java.util.List;
 import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Collectors;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.lucene.search.join.ScoreMode;
 import org.elasticsearch.common.collect.Tuple;
 import org.elasticsearch.index.query.BoolQueryBuilder;
 import org.elasticsearch.index.query.MatchQueryBuilder;
 import org.elasticsearch.index.query.NestedQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
 import org.springframework.stereotype.Component;
 
 @Component
 public class ProgrammeBasedFilterStrategy implements RoleBasedFilterStrategy {
 
-  private static final String COLUMN_FILTER = "programmeName";
+  private static final String COLUMN_FILTER = "programmeId";
 
   @Override
-  public Optional<Tuple<String, BoolQueryBuilder>> getFilter() {
+  public Optional<Tuple<String, BoolQueryBuilder>> getFilter(List<ColumnFilter> columnFilters) {
+    MatchQueryBuilder statusMatchQueryBuilder = null;
+    if (columnFilters != null) {
+      Set<ColumnFilter> statusFilters = columnFilters.stream().filter(
+              columnFilter -> StringUtils.equals(columnFilter.getName(), "programmeMembershipStatus"))
+          .collect(
+              Collectors.toSet());
+      if (statusFilters.size() > 0) {
+        ProgrammeMembershipStatus status = ProgrammeMembershipStatus
+            .valueOf(statusFilters.iterator().next().getValues().get(0).toString());
+        statusMatchQueryBuilder = getProgrammeMembershipStatusQueryBuilder(status);
+      }
+    }
+
     UserProfile currentUserProfile = TisSecurityHelper.getProfileFromContext();
     Set<Programme> assignedProgrammes = currentUserProfile.getAssignedProgrammes();
     if (CollectionUtils.isNotEmpty(assignedProgrammes)) {
       BoolQueryBuilder programmeRoleFilter = new BoolQueryBuilder();
-      assignedProgrammes.forEach(programme -> programmeRoleFilter.should(
-          new NestedQueryBuilder("programmeMemberships",
-              new MatchQueryBuilder("programmeMemberships.programmeId", programme.getId()),
-              ScoreMode.None)));
+      for (Programme programme : assignedProgrammes) {
+        if (statusMatchQueryBuilder == null) {
+          programmeRoleFilter.should(
+              new NestedQueryBuilder("programmeMemberships",
+                  new MatchQueryBuilder("programmeMemberships.programmeId", programme.getId()),
+                  ScoreMode.None));
```

Comment:
There is some duplication here that might improve maintainability here by making the intention clearer.  I'm happy enough if you want to leave this.

Reply:
I prefer to leaving as what it is.